js/src/views/Signer/components/RequestOrigin/requestOrigin.js

```diff
@@ -29,7 +29,9 @@ export default class RequestOrigin extends Component {
   static propTypes = {
     origin: PropTypes.shape({
       type: PropTypes.oneOf(['unknown', 'dapp', 'rpc', 'ipc', 'signer']),
-      details: PropTypes.string.isRequired
+      details: PropTypes.shape({
```

Comment:
Actually it can be either string (RPC origin, Dapp id, IPC session) or object with `details` in it.